app/controllers/admin/orders_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Admin::OrdersController < Admin::AdminController
+  before_action :load_orders, only: :index
+  before_action :load_bill, only: :update
+
+  def index; end
+
+  def update
+    ActiveRecord::Base.transaction do
+      handle_status
+      redirect_to admin_orders_path status: @bill.status
+    end
```

Comment:
```suggestion      
    end
    redirect_to admin_orders_path status: @bill.status
```
redirect thì ko cần bỏ vào transaction

Chị thấy em đang đặt `ActiveRecord::Base.transaction do` hơi thừa 

Ví dụ 
```
 unless @bill.allowed_to_cancel?
      flash[:danger] = t "pages.admin.update.order.status.cancel_error",
                         status: t("pages.bill.status.#{@bill.status}").downcase
      return
    end
    ```
    thì cần j transaction

Reply:
Chị nghĩ code này chỉ cần
```
def update
    handle_status
    redirect_to admin_orders_path status: @bill.status

rescue ActiveRecord::RecordNotFound
    flash[:danger] = t "pages.bill.not_found"
    redirect_to admin_orders_path status: @bill.status
  end
```

Reply:
còn transaction đặt những chỗ cần 
ví dụ ở dưới

Reply:
dạ vâg ạ